src/poetry/puzzle/provider.py

```diff
@@ -469,7 +469,14 @@ def complete_package(
                 if extra not in package.extras:
                     continue
 
-                optional_dependencies += [d.name for d in package.extras[extra]]
+                for extra_dependency in package.extras[extra]:
+                    if extra_dependency.name == dependency.name:
+                        # If an extra/feature refers to another in the base package,
+                        # resolve the dependencies of that extra/feature.
+                        for feature in extra_dependency.features:
+                            for pd in package.extras.get(feature, []):
+                                _dependencies.append(pd)
```

Comment:
I think this adds `pd` to `_dependencies` too soon.  The code a few lines below will check markers etc before copying from `requires` to `_dependencies`, this skips any such checks

Reply:
I think there is likely also a problem later, in the block `# For normal dependency resolution, we have to make sure that root extras ...`

`dep.in_extras` only reflects the the "direct" extras that require `dep`.  But we should now also understand the top-level extras that might require `dep` through the indirection

Reply:
Seems we do not encounter this case due to https://github.com/python-poetry/poetry/blob/603646c64601eb49886a97f6e930aba2cc520475/src/poetry/packages/locker.py#L577.

Which i suspect is a bug in itself. Seems [#6237](https://github.com/python-poetry/poetry/commit/5f2d2b866379a6f4f6779d382fde8aee9a2c8498#diff-b0671e034884534eca80d94078a8d744a370d79b9cab043257523440ba887f62R413) swallowed a [TODO](https://github.com/python-poetry/poetry/blob/9fb6c95d3784b79891747a2fbf05a1b95e1aa138/src/poetry/packages/locker.py#L413C45-L414) we had.

Reply:
#10119